lxd/instance/drivers/driver_qemu.go

```diff
@@ -1704,6 +1707,10 @@ func (vm *qemu) generateQemuConfigFile(busName string, devConfs []*deviceConfig.
 
 	// Record the mounts we are going to do inside the VM using the agent.
 	agentMounts := []instancetype.VMAgentMount{}
+
+	// These devices are sorted so that NICs are added first to ensure that the first NIC can use the 5th
```

Comment:
@stgraber same concern here as https://github.com/lxc/lxd/pull/7525/files#r439429739

Reply:
Yeah, that's only true on PCIe but even on non-PCIe, having NICs first means that their names won't change when you add a disk, gpu, ...